src/ray/common/event_stats.cc

```diff
@@ -81,8 +82,26 @@ std::shared_ptr<StatsHandle> EventTracker::RecordStart(
       global_stats_);
 }
 
+void EventTracker::RecordEnd(std::shared_ptr<StatsHandle> handle) {
+  RAY_CHECK(!handle->end_or_execution_recorded);
```

Comment:
Should we just make it no-op instead? Seems too hard check for just instrumentation library

Reply:
Added a comment in the docstring saying the pair of methods must be called.